### ISSUE TITLE
More options for mirrors

### ISSUE THREAD

**Issue by lord2800** [gds-operations/puppet-aptly#35]
I'd like support for the `-with-udebs` and `-with-sources` flags. Support for `-filter` and `-filter-with-deps` would also be nice.

**Comment by antonio** [comment]
`-with-udebs` and `-with-sources` are already supported with parameters. With regards to other parameters and flags, I have a branch that addresses this, but haven't opened a PR since it is disruptive and will break other people's manifests. 

It drops the flags and instead adds a hash parameter in which you can specify any flag or option for `aptly`. I use it to specify the parameters via hiera, like this:

```yml
  ntop-precise-all:
    location: 'http://packages.ntop.org/apt-stable/12.04/'
    key: '7921DF34'
    release: 'precise'
    prefix: 'ntop-precise-all'
    release_path: 'all/'
    mirror_cmd_opts:
      '-with-sources': false
      '-force-components': true
```

If you are interested, you can find it at https://github.com/antonio/puppet-aptly/tree/cmd-options

Also, if any of the maintainers of the project is interested in this as well, I can update my branch to add back the parameters I've delete (`with-sources` and `with-udebs`) and add a deprecation notice to them, and open a PR with my changes.

**Comment by lord2800** [comment]
Ah ha! I'm using 0.3.0, which doesn't have support for that (and other things). Can a new version be tagged, then?

**Comment by antonio** [comment]
@lord2800 are you using librarian-puppet by any chance? If so, you can just update your `Puppetfile` to point to my branch (which is 0.3.0 plus my patch) with

```
mod "aptly",               :git => "git://github.com/antonio/puppet-aptly", :ref => "cmd-options"
```

**Comment by mattbostock** [comment]
We'll release a new version as soon as we can, thanks for highlighting this.

**Comment by lord2800** [comment]
@antonio I am, but I prefer to use official sources whenever possible.

**Comment by alexmuller** [comment]
We just released 0.4.0 which includes the changes to `aptly::mirror`, sorry for the delay.
